services/web-server/test/graphql/tasks_graphql_test.js

```diff
@@ -64,4 +68,81 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
       assert.equal(response.data.createTask.taskId, taskId);
     });
   });
+
+  suite('Task Subscriptions', function() {
+    helper.withMockedEventIterator();
+
+    const createSubscriptionClient = async () => {
+      return new Promise(function(resolve, reject) {
+        const subscriptionClient = new SubscriptionClient(
+          `ws://localhost:${helper.serverPort}/subscription`,
+          {
+            reconnect: true,
+          },
+          WebSocket,
+        );
+        subscriptionClient.onConnected(function() {
+          resolve(subscriptionClient);
+        });
+        subscriptionClient.onError(function(err) {
+          reject(err);
+        });
+      });
+    };
+
+    const getClient = (subscriptionClient) => {
+      const cache = new InMemoryCache();
+      const link = new WebSocketLink(subscriptionClient);
+
+      return new ApolloClient({ cache, link });
+    };
+
+    test('subscribe works', async function(){
+      // We need to create this subscription client separately so we can close it after our test
+      // Otherwise, our tests will just hang and timeout
+      let subscriptionClient = await createSubscriptionClient();
+      const client = getClient(subscriptionClient);
+      const task = helper.makeTaskDefinition();
+      const taskGroupId = task.taskGroupId;
+      const subscriptions = [
+        'tasksDefined',
+      ];
+
+      const payload = {
+        tasksSubscriptions: {
+          status: {
+            taskId: "subscribe-task-id",
+            taskGroupId: "subscribe-task-group-id",
+          },
+        },
+      };
+
+      const asyncIterator = new Object();
+      asyncIterator[Symbol.asyncIterator] = async function*() {
+        yield payload;
+      };
+
+      helper.SetNextAsyncIterator(asyncIterator);
+
+      let tasksSubscriptionsResult;
+      let taskSubscription = client.subscribe({
+        query: gql`${subscribeTasks}`,
+        variables: {
+          taskGroupId,
+          subscriptions,
```

Comment:
So, this could be simplified a bit to 
```suggestion
          taskGroupId: "subscribe-task-group-id",
          subscriptions: ['tasksDefined'],
```